root/static/scripts/common/components/Autocomplete2/formatters.js

```diff
@@ -0,0 +1,433 @@
+/*
+ * @flow
+ * Copyright (C) 2019 MetaBrainz Foundation
+ *
+ * This file is part of MusicBrainz, the open internet music database,
+ * and is licensed under the GPL version 2, or (at your option) any
+ * later version: http://www.gnu.org/licenses/gpl-2.0.txt
+ */
+
+import * as React from 'react';
+
+import {commaOnlyListText} from '../../i18n/commaOnlyList';
+import {unwrapNl} from '../../i18n';
+import {addColonText} from '../../i18n/addColon';
+import {reduceArtistCredit} from '../../immutable-entities';
+import bracketed, {bracketedText} from '../../utility/bracketed';
+import formatDate from '../../utility/formatDate';
+import formatDatePeriod from '../../utility/formatDatePeriod';
+import formatTrackLength from '../../utility/formatTrackLength';
+import primaryAreaCode from '../../utility/primaryAreaCode';
+
+import type {
+  AutocompleteAreaT,
+  AutocompleteEventT,
+  AutocompleteInstrumentT,
+  AutocompletePlaceT,
+  AutocompleteRecordingT,
+  AutocompleteReleaseT,
+  AutocompleteReleaseGroupT,
+  AutocompleteSeriesT,
+  AutocompleteWorkT,
+  Item,
+} from './types';
+
+const nonLatinRegExp = /[^\u0000-\u02ff\u1E00-\u1EFF\u2000-\u207F]/;
+
+function isLatin(str) {
+  return !nonLatinRegExp.test(str);
+}
+
+function showComment(children, className = 'comment') {
+  return (
+    <span className={`autocomplete-${className}`}>
+      {children}
+    </span>
+  );
+}
```

Comment:
Renamed it to `showExtraInfo` (and renamed `showLineComment` to `showExtraInfoLine`).

Reply:
> I was also wondering why functions rather than components, but tbh I'm still not sure when to use one over the other sometimes.

There'd be nothing technically wrong with always using components, but I sometimes just prefer to use functions in cases where they do something very simple (wrapping something in a span) and are used heavily as a utility to the point where the component syntax would become a burden. Another case I'd use a function is if it's being used as a map callback, like in `release.events.map(formatReleaseEvent)`.